src/include/api.h

```diff
@@ -116,6 +116,14 @@
 	API_CALL_NOCONF(s, WT_CURSOR, n, cur,				\
 	    ((bt) == NULL) ? NULL : ((WT_BTREE *)(bt))->dhandle)
 
+#define	JOINABLE_CURSOR_CALL_CHECK(cur)					\
+	if (F_ISSET(cur, WT_CURSTD_JOINED))				\
+		WT_ERR(__wt_curindex_joined(cur))
+
+#define	JOINABLE_CURSOR_API_CALL(cur, s, n, bt)				\
+	CURSOR_API_CALL(cur, s, n, bt);					\
+	JOINABLE_CURSOR_CALL_CHECK(cur)
+
```

Comment:
Did you consider adding a `JOINABLE_CURSOR_UPDATE_API_CALL`?

I know it's a mouthful, but I think it would save exporting `JOINABLE_CURSOR_CALL_CHECK`.  If you wanted to call these `CURJOIN_API_CALL` and `CURJOIN_UPDATE_API_CALL`, I wouldn't object.